File: 프로그래머스/0/181832. 정수를 나선형으로 배치하기/정수를 나선형으로 배치하기.cs
using System;

public class Solution 
{
    public int[,] solution(int n) 
    {
        int[,] nArr = new int[n,n];
        
        int count = 1;
        
        int colMax = n - 1; // 오른쪽 경계
        int colMin = 0; // 왼쪽 경계
        int rowMax = n - 1; // 아래쪽 경계
        int rowMin = 0; // 위쪽 경계
        while(count <= n * n)
        {
            for(int j = colMin; j <= colMax; j++)
            {
                nArr[rowMin,j] = count;
                count++;
            }
            rowMin++;  // 위쪽 경계를 한칸 내림
            for(int i = rowMin; i <= rowMax; i++)
            {
                nArr[i,colMax] = count;
                count++;
            }
            colMax--; // 오른족 경계를 왼쪽으로 한칸 댕김
            for(int j = colMax; j>= colMin; j--)
            {
                nArr[rowMax,j] = count;
                count++;
            }
            rowMax--; // 아래쪽 경계를 위로 한칸 올림
            for(int i = rowMax; i >= rowMin; i--)
            {
                nArr[i,colMin] = count;
                count++;
            }
            colMin++; // 왼쪽 경계를 오른족으로 한칸 댕김      
        }
        return nArr;
    }
}

File: 프로그래머스/0/181832. 정수를 나선형으로 배치하기/정수를 나선형으로 배치하기.cpp
#include <string>
#include <vector>

using namespace std;

vector<vector<int>> solution(int n) 
{
    vector<vector<int>> answer(n, vector<int>(n,0));
    
    int rowMax = n-1;
    int colMax = n-1;
    int rowMin = 0;
    int colMin = 0;
    
    int i = 1;
    while(i <= n * n)
    {
        for(int c=colMin; c<=colMax; c++)
        {
            answer[rowMin][c] = i;
            i++;
        }
        rowMin++;
        for(int r=rowMin; r<=rowMax; r++)
        {
            answer[r][colMax] = i;
            i++;
        }
        colMax--;
        for(int c=colMax; c>=colMin; c--)
        {
            answer[rowMax][c] = i;
            i++;
        }
        rowMax--;
        for(int r=rowMax; r>=rowMin; r--)
        {
            answer[r][colMin] = i;
            i++;
        }
        colMin++;
    }
    return answer;
}